phoenix-core/src/main/java/org/apache/phoenix/compile/QueryPlan.java

```diff
@@ -59,6 +60,8 @@
      * Returns projector used to formulate resultSet row
      */
     RowProjector getProjector();
+
+    String getCursorName();
```

Comment:
I think an alternate implementation would be to have a CursorQueryPlan which wraps another QueryPlan. This would be similar to the way a UnionPlan was implemented. That way you wouldn't need to litter the other plan implementations with logic to wrap them with the CursorResultIterator (which would be brittle if/when new plans are added).